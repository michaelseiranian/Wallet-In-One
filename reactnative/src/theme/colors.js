// Colours used across different components of the app when 'light' theme is enabled in settings.
export const lightColors = {
    background: 'white',
    text: 'black',
    primary: 'purple',  // Main theme colour of the app
    secondary: 'blue',
    stock_account: 'white', // Used for the background of the main stocks screen
};

// Colours used across different components of the app when 'dark' theme is enabled in settings.
export const darkColors = {
    background: '#242526',
    text: 'white',
    primary: 'purple',  // Main theme colour of the app
    secondary: 'blue',
    stock_account: '#181818', // Used for the background of the main stocks screen
};