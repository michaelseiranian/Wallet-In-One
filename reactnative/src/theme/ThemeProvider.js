import React, {createContext, useEffect, useContext, useState, AsyncStorage} from 'react';
import {lightColors, darkColors} from './colors';
import {useColorScheme} from 'react-native';
export const ThemeContext = createContext({
    dark: false,
    colors: lightColors,
    setScheme: () => {},
    update: () => {}
});

import * as SecureStore from 'expo-secure-store';

export const ThemeProvider = props => {
    const defaultColorScheme = useColorScheme(); // colorScheme is either 'dark', 'light' or 'null'
    const [isDark, setIsDark] = useState(false);

    async function loadDarkModeSettings(value = 'default') {
        // colorScheme is either 'dark', 'light' or 'null'
        var darkSetting;
        if (value === "default"){
            darkSetting = await SecureStore.getItemAsync('darkModeSettings');
        }
        else{
            darkSetting = value;
        }

        console.log('settings, default: ', darkSetting, defaultColorScheme,value)

        if (darkSetting === 'true') {
            setIsDark(true)
        }
        else if (darkSetting === 'false') {
            setIsDark(false)
        }
        else if (defaultColorScheme == 'dark') {
            setIsDark(true)
        }
        else if (defaultColorScheme == 'light') {
            setIsDark(false)
        }
        else {
            setIsDark(false)
        }
    }

    // Load Dark Mode Setting
    useEffect(() => {    
        loadDarkModeSettings();
    }, [defaultColorScheme]);

    const defaultTheme = {
        dark: isDark,
        colors: isDark ? darkColors : lightColors,
        setScheme: scheme => setIsDark(scheme === 'dark'),
        update: loadDarkModeSettings,
    };
    return (
        <ThemeContext.Provider value={defaultTheme}>
            {props.children}
        </ThemeContext.Provider>
    );
};

// creating a custom hook for accessing all the values

export const useTheme = () => useContext(ThemeContext);