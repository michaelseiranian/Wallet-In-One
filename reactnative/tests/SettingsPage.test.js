import React from 'react';
import SettingsPage from 'reactnative/screens/SettingsPage';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent, act } from '@testing-library/react-native';
import { useTheme } from 'reactnative/src/theme/ThemeProvider';
import * as SecureStore from "expo-secure-store";
jest.mock("expo-secure-store");
jest.mock('reactnative/src/theme/ThemeProvider');

SecureStore.getItemAsync.mockReset();
useTheme.mockReturnValue({
  dark: false,
  colors: {
    text: '#000',
    background: '#fff',
    primary: '#0af',
  },
  setScheme: jest.fn(),
  update: jest.fn(),
});

jest.mock('react', () => {
  const ActualReact = jest.requireActual('react')
  return {
    ...ActualReact,
    useContext: () => (ActualReact.useState({ 'signedIn': false })),
  }
})

it('snapshot test for settings page', () => {
  const navigate = jest.fn();
  const settingsPage = renderer.create(<SettingsPage navigation={{ navigate }} />);
  expect(settingsPage).toMatchSnapshot();
});

it('settings page pressing log out button test', () => {
  const navigate = jest.fn();
  const settingsPage = render(<SettingsPage navigation={{ navigate }} />);
  act(() => {
    fireEvent.press(screen.getByText('Logout'));
  });
  expect(settingsPage).toMatchSnapshot();
});


it('setting page dark mode toggle', () => {
  const navigate = jest.fn();
  const settingsPage = render(<SettingsPage navigation={{ navigate }} />);
  fireEvent.press(screen.getByText('Dark Mode'));
  expect(settingsPage).toMatchSnapshot();
});

it('setting page system default toggle', () => {
  const navigate = jest.fn();
  const settingsPage = render(<SettingsPage navigation={{ navigate }} />);
  fireEvent.press(screen.getByText('System Default'));
  expect(settingsPage).toMatchSnapshot();
});

it('setting page light mode toggle', () => {
  const navigate = jest.fn();
  const settingsPage = render(<SettingsPage navigation={{ navigate }} />);
  fireEvent.press(screen.getByText('Light Mode'));
  expect(settingsPage).toMatchSnapshot();
});