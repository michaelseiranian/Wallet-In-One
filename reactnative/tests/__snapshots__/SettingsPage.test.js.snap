// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setting page dark mode toggle 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "center",
      "flexGrow": 1,
      "justifyContent": "center",
      "paddingBottom": 20,
    }
  }
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        [
          {
            "color": "#000",
            "fontWeight": "700",
          },
          {
            "fontSize": 24,
            "marginBottom": 20,
          },
        ]
      }
    >
      Notifications
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "80%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#000",
          }
        }
      >
        Receive notifications
      </Text>
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="#81b0ff"
        style={
          {
            "height": 31,
            "width": 51,
          }
        }
        thumbTintColor="purple"
        tintColor="#767577"
        value={true}
      />
    </View>
    <Text
      style={
        [
          {
            "color": "#000",
            "fontWeight": "700",
          },
          {
            "fontSize": 24,
            "marginBottom": 20,
          },
        ]
      }
    >
      Themes
    </Text>
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
            "borderColor": "lightgrey",
            "borderRadius": 4,
            "borderWidth": 1,
            "width": "80%",
          },
        ]
      }
    >
      <View
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderBottomColor": "#ddd",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "space-between",
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
              }
            }
          >
            System Default
          </Text>
        </View>
      </View>
      <View
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderBottomColor": "#ddd",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "space-between",
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
              }
            }
          >
            Light Mode
          </Text>
        </View>
      </View>
      <View
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderBottomColor": "#ddd",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "padding": 10,
              },
              {
                "borderBottomWidth": 0,
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "#000",
              }
            }
          >
            Dark Mode
          </Text>
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "green",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            ï„«
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          [
            {
              "backgroundColor": "#0af",
            },
            {
              "color": "white",
            },
            {
              "borderRadius": 25,
              "fontSize": 20,
              "fontWeight": "bold",
              "marginTop": "4%",
              "paddingHorizontal": "12%",
              "paddingVertical": "2%",
              "textAlign": "center",
              "width": "75%",
            },
          ]
        }
      >
        Logout
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`setting page light mode toggle 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "center",
      "flexGrow": 1,
      "justifyContent": "center",
      "paddingBottom": 20,
    }
  }
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        [
          {
            "color": "#000",
            "fontWeight": "700",
          },
          {
            "fontSize": 24,
            "marginBottom": 20,
          },
        ]
      }
    >
      Notifications
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "80%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#000",
          }
        }
      >
        Receive notifications
      </Text>
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="#81b0ff"
        style={
          {
            "height": 31,
            "width": 51,
          }
        }
        thumbTintColor="purple"
        tintColor="#767577"
        value={true}
      />
    </View>
    <Text
      style={
        [
          {
            "color": "#000",
            "fontWeight": "700",
          },
          {
            "fontSize": 24,
            "marginBottom": 20,
          },
        ]
      }
    >
      Themes
    </Text>
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
            "borderColor": "lightgrey",
            "borderRadius": 4,
            "borderWidth": 1,
            "width": "80%",
          },
        ]
      }
    >
      <View
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderBottomColor": "#ddd",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "space-between",
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
              }
            }
          >
            System Default
          </Text>
        </View>
      </View>
      <View
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderBottomColor": "#ddd",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "space-between",
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
              }
            }
          >
            Light Mode
          </Text>
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "green",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            ï„«
          </Text>
        </View>
      </View>
      <View
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderBottomColor": "#ddd",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "padding": 10,
              },
              {
                "borderBottomWidth": 0,
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "#000",
              }
            }
          >
            Dark Mode
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          [
            {
              "backgroundColor": "#0af",
            },
            {
              "color": "white",
            },
            {
              "borderRadius": 25,
              "fontSize": 20,
              "fontWeight": "bold",
              "marginTop": "4%",
              "paddingHorizontal": "12%",
              "paddingVertical": "2%",
              "textAlign": "center",
              "width": "75%",
            },
          ]
        }
      >
        Logout
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`setting page system default toggle 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "center",
      "flexGrow": 1,
      "justifyContent": "center",
      "paddingBottom": 20,
    }
  }
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        [
          {
            "color": "#000",
            "fontWeight": "700",
          },
          {
            "fontSize": 24,
            "marginBottom": 20,
          },
        ]
      }
    >
      Notifications
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "80%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#000",
          }
        }
      >
        Receive notifications
      </Text>
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="#81b0ff"
        style={
          {
            "height": 31,
            "width": 51,
          }
        }
        thumbTintColor="purple"
        tintColor="#767577"
        value={true}
      />
    </View>
    <Text
      style={
        [
          {
            "color": "#000",
            "fontWeight": "700",
          },
          {
            "fontSize": 24,
            "marginBottom": 20,
          },
        ]
      }
    >
      Themes
    </Text>
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
            "borderColor": "lightgrey",
            "borderRadius": 4,
            "borderWidth": 1,
            "width": "80%",
          },
        ]
      }
    >
      <View
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderBottomColor": "#ddd",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "space-between",
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
              }
            }
          >
            System Default
          </Text>
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "green",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            ï„«
          </Text>
        </View>
      </View>
      <View
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderBottomColor": "#ddd",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "space-between",
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
              }
            }
          >
            Light Mode
          </Text>
        </View>
      </View>
      <View
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderBottomColor": "#ddd",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "padding": 10,
              },
              {
                "borderBottomWidth": 0,
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "#000",
              }
            }
          >
            Dark Mode
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          [
            {
              "backgroundColor": "#0af",
            },
            {
              "color": "white",
            },
            {
              "borderRadius": 25,
              "fontSize": 20,
              "fontWeight": "bold",
              "marginTop": "4%",
              "paddingHorizontal": "12%",
              "paddingVertical": "2%",
              "textAlign": "center",
              "width": "75%",
            },
          ]
        }
      >
        Logout
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`settings page pressing log out button test 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "center",
      "flexGrow": 1,
      "justifyContent": "center",
      "paddingBottom": 20,
    }
  }
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        [
          {
            "color": "#000",
            "fontWeight": "700",
          },
          {
            "fontSize": 24,
            "marginBottom": 20,
          },
        ]
      }
    >
      Notifications
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "80%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#000",
          }
        }
      >
        Receive notifications
      </Text>
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="#81b0ff"
        style={
          {
            "height": 31,
            "width": 51,
          }
        }
        thumbTintColor="purple"
        tintColor="#767577"
        value={true}
      />
    </View>
    <Text
      style={
        [
          {
            "color": "#000",
            "fontWeight": "700",
          },
          {
            "fontSize": 24,
            "marginBottom": 20,
          },
        ]
      }
    >
      Themes
    </Text>
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
            "borderColor": "lightgrey",
            "borderRadius": 4,
            "borderWidth": 1,
            "width": "80%",
          },
        ]
      }
    >
      <View
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderBottomColor": "#ddd",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "space-between",
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
              }
            }
          >
            System Default
          </Text>
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "green",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            ï„«
          </Text>
        </View>
      </View>
      <View
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderBottomColor": "#ddd",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "space-between",
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
              }
            }
          >
            Light Mode
          </Text>
        </View>
      </View>
      <View
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderBottomColor": "#ddd",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "padding": 10,
              },
              {
                "borderBottomWidth": 0,
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "#000",
              }
            }
          >
            Dark Mode
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          [
            {
              "backgroundColor": "#0af",
            },
            {
              "color": "white",
            },
            {
              "borderRadius": 25,
              "fontSize": 20,
              "fontWeight": "bold",
              "marginTop": "4%",
              "paddingHorizontal": "12%",
              "paddingVertical": "2%",
              "textAlign": "center",
              "width": "75%",
            },
          ]
        }
      >
        Logout
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`snapshot test for settings page 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "center",
      "flexGrow": 1,
      "justifyContent": "center",
      "paddingBottom": 20,
    }
  }
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        [
          {
            "color": "#000",
            "fontWeight": "700",
          },
          {
            "fontSize": 24,
            "marginBottom": 20,
          },
        ]
      }
    >
      Notifications
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "80%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#000",
          }
        }
      >
        Receive notifications
      </Text>
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="#81b0ff"
        style={
          {
            "height": 31,
            "width": 51,
          }
        }
        thumbTintColor="purple"
        tintColor="#767577"
        value={true}
      />
    </View>
    <Text
      style={
        [
          {
            "color": "#000",
            "fontWeight": "700",
          },
          {
            "fontSize": 24,
            "marginBottom": 20,
          },
        ]
      }
    >
      Themes
    </Text>
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
            "borderColor": "lightgrey",
            "borderRadius": 4,
            "borderWidth": 1,
            "width": "80%",
          },
        ]
      }
    >
      <View
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderBottomColor": "#ddd",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "space-between",
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
              }
            }
          >
            System Default
          </Text>
          <Text />
        </View>
      </View>
      <View
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderBottomColor": "#ddd",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "space-between",
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
              }
            }
          >
            Light Mode
          </Text>
        </View>
      </View>
      <View
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderBottomColor": "#ddd",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "padding": 10,
              },
              {
                "borderBottomWidth": 0,
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "#000",
              }
            }
          >
            Dark Mode
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          [
            {
              "backgroundColor": "#0af",
            },
            {
              "color": "white",
            },
            {
              "borderRadius": 25,
              "fontSize": 20,
              "fontWeight": "bold",
              "marginTop": "4%",
              "paddingHorizontal": "12%",
              "paddingVertical": "2%",
              "textAlign": "center",
              "width": "75%",
            },
          ]
        }
      >
        Logout
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
